src/EnergyPlus/SystemAvailabilityManager.cc

```diff
@@ -602,182 +582,162 @@ namespace SystemAvailabilityManager {
-                        state.dataSystemAvailabilityManager->NCycSysAvailMgrData(SysAvailNum).CtrlType = CycleOnAnyCoolingZone;
-                    } else if (SELECT_CASE_var == "CYCLEONANYHEATINGZONE") {
-                        state.dataSystemAvailabilityManager->NCycSysAvailMgrData(SysAvailNum).CtrlType = CycleOnAnyHeatingZone;
-                    } else if (SELECT_CASE_var == "CYCLEONANYHEATINGZONEFANSONLY") {
-                        state.dataSystemAvailabilityManager->NCycSysAvailMgrData(SysAvailNum).CtrlType = CycleOnAnyHeatingZoneFansOnly;
-                    } else {
-                        ShowSevereError(state, std::string{RoutineName} + cCurrentModuleObject + "=\"" + cAlphaArgs(1) + "\", invalid");
-                        ShowSevereError(state, std::string{RoutineName} + "incorrect value: " + cAlphaFieldNames(4) + "=\"" + cAlphaArgs(4) + "\".");
-                        ErrorsFound = true;
-                    }
-                }
+                nightCycleMgr.nightCycleControlType = static_cast<NightCycleControlType>(
+                    getEnumerationValue(NightCycleControlTypeNamesUC, UtilityRoutines::MakeUPPERCase(cAlphaArgs(4))));
+
+                assert(nightCycleMgr.nightCycleControlType != NightCycleControlType::Invalid);
 
                 // Cycling Run Time Control Type
-                if (!lAlphaFieldBlanks(5)) {
-                    {
-                        auto const SELECT_CASE_var(UtilityRoutines::MakeUPPERCase(cAlphaArgs(5)));
-                        if (SELECT_CASE_var == "FIXEDRUNTIME") {
-                            state.dataSystemAvailabilityManager->NCycSysAvailMgrData(SysAvailNum).CycRunTimeCntrlType = FixedRunTime;
-                        } else if (SELECT_CASE_var == "THERMOSTAT") {
-                            state.dataSystemAvailabilityManager->NCycSysAvailMgrData(SysAvailNum).CycRunTimeCntrlType = Thermostat;
-                        } else if (SELECT_CASE_var == "THERMOSTATWITHMINIMUMRUNTIME") {
-                            state.dataSystemAvailabilityManager->NCycSysAvailMgrData(SysAvailNum).CycRunTimeCntrlType = ThermostatWithMinimumRunTime;
-                        } else {
-                            ShowSevereError(state, std::string{RoutineName} + cCurrentModuleObject + "=\"" + cAlphaArgs(1) + "\", invalid");
-                            ShowSevereError(state,
-                                            std::string{RoutineName} + "incorrect value: " + cAlphaFieldNames(5) + "=\"" + cAlphaArgs(5) + "\".");
-                            ErrorsFound = true;
-                        }
-                    }
-                }
+                nightCycleMgr.cyclingRunTimeControl = static_cast<CyclingRunTimeControl>(
+                    getEnumerationValue(CyclingRunTimeControlNamesUC, UtilityRoutines::MakeUPPERCase(cAlphaArgs(5))));
+
+                assert(nightCycleMgr.cyclingRunTimeControl != CyclingRunTimeControl::Invalid);
```

Comment:
Same

Reply:
Verified. A5 of `AvailabilityManager:NightCycle` (Cycling Run Time Control Type) is of type choice in `Energy+.idd.in`.